src/msg/simple/SimpleMessenger.cc

```diff
@@ -357,6 +357,7 @@ Pipe *SimpleMessenger::connect_rank(const entity_addr_t& addr,
 			static_cast<PipeConnection*>(con));
   pipe->pipe_lock.Lock();
   pipe->set_peer_type(type);
+  pipe->set_peer_id(num);
```

Comment:
I found we only set peer id when connecting, what if the connection is accepted to build?